File: 프로그래머스/lv1/42748. K번째수/K번째수.py
def solution(array, commands):
    result = []
    for cmd in commands:
        i, j, k = cmd
        # 인덱스 보정
        i -= 1
        j -= 1
        k -= 1

        # 슬라이스 생성
        slice = array[i:j+1]
        slice.sort()
        result.append(slice[k])

    return result

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int [] answer = new int[commands.length];
                
        for(int i = 0; i < commands.length; i++){
            int start = commands[i][0];
            int end = commands[i][1];
            int index = commands[i][2];
            
            List<Integer> lst = new ArrayList<>();
            
            for(int j = start; j <= end; j++){
                lst.add(array[j-1]);
            }
            Collections.sort(lst);
            answer[i] = lst.get(index-1);
        }
        
        return answer;
    }
}